### PR TITLE
Converted update process to rely on HandlerManifest

### PR DESCRIPTION
**Pull request by brendandixon**
Signed-off-by: Brendan Dixon <brendand@microsoft.com>

### CODE REVIEW

**Comment by ahmetalpbalkan** [comment]
can we create the .zip on the fly? I'm not a big fan of checking in binaries to source tree.

**Comment by brendandixon** [comment]
@ahmetalpbalkan Not for now. It's best if left as it is. Actually checking in a binary is a good thing in this case. Otherwise we'd need to ensure (somehow) the test binary really matches that which we produce for production. By checking in a production binary we know they're the same.

**Comment by brendandixon** [comment]
@ahmetalpbalkan @hglkrijger Can we merge this?

File: azurelinuxagent/ga/update.py [hunk 1]
```diff
@@ -100,6 +120,7 @@ def run_latest(self):
                 latest_agent.mark_failure(is_fatal=True)
 
         sys.exit(0)
+        return
```
Comment (hglkrijger) [inline azurelinuxagent/ga/update.py]: is this return necessary?

**Comment by hglkrijger** [comment]
lgtm

**Pull request merged**